infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -953,6 +955,20 @@ public void append(List<LogData> range) {
         SegmentHandle firstSh = getSegmentHandleForAddress(first.getGlobalAddress());
         SegmentHandle lastSh = getSegmentHandleForAddress(last.getGlobalAddress());
 
+        // Extract all addresses associated with the provided write range.
+        Set<Long> pendingWrite = range.stream()
+                .map(entry -> entry.getGlobalAddress()).collect(Collectors.toSet());
+
+        // See if the provided range overlaps with any of the previously written entries.
+        Set<Long> segOneOverlap = Sets.intersection(pendingWrite,
+                firstSh.getKnownAddresses().keySet());
+        Set<Long> segTwoOverlap = Sets.intersection(pendingWrite,
+                lastSh.getKnownAddresses().keySet());
+        if (!segOneOverlap.isEmpty() || !segTwoOverlap.isEmpty()) {
+            log.error("Overlapping addresses detected: {}, {}", segOneOverlap, segTwoOverlap);
+            throw new OverwriteException(OverwriteCause.NONE);
```

Comment:
I don't think this makes sense:
/** Indicates there is no actual cause for the overwrite, hence no overwrite exception has occurred. */
NONE(4);

Reply:
The problem here is that there is really no suitable "cause" for this overwrite, mostly because of the N-to-1 mapping. Do we specify an overwrite cause for the first conflicting entry? Last? In any case, this will not give us any useful information.

Reply:
You can actually map it to either a duplicate, same write, or a hole. It won't give us more information, but for the sake of consistency. 

You can actually remove the OverwriteCause enum, Its used for an optimization, which I don't think is a reasonable one.

Reply:
Done.